vehicle/vw/id/api.go

```diff
@@ -91,13 +96,12 @@ func (v *API) Action(vin, action, value string) error {
 	uri := fmt.Sprintf("%s/vehicles/%s/%s/%s", BaseURL, vin, action, value)
 
 	req, err := request.New(http.MethodPost, uri, nil, request.AcceptJSON)
-
```

Comment:
Dito- das hat mit dem PR nichts zu tun?

Reply:
> I've changed the ID data handling for the different endpoint and made error handling more go like with quick return 

Ups. Das möchte ich eigentlich nicht ändern, zumal es im Rest der Codebase auch so ist.

Reply:
Ganz ehrlich, dieses `err == nil` checken ist eigentlich überhaupt nicht go like und macht den code nicht besser sondern eher schlechter lesbar. Ich verstehe wenn es in der restlichen codebasis evtl überall so ist, aber das machts nicht besser. 

default go ist halt

```
if err != nil{
   return err
}
```

Ich denke wenn es pro file consistent ist kann man das schon machen.

Reply:
Auch die OpenAI findet go error handling mit quick return passender. 😉 

<img width="780" alt="image" src="https://user-images.githubusercontent.com/5090631/205910573-ad8c2917-398f-4544-bbb3-feb207314ca0.png">

Reply:
Yep, aber das ist ein anderer PR.

Reply:
`if err := ...; err != nil ...`

Diese Schreibweise eignet sich nur bei single return error func. 

```
err, value = foo()
if err != nil {
  return err
}
```

☝️ Das ist best practice Go error handling und findest in jeder codebasis genau so.

Ist auch kein problem da es mit `nil` überschrieben wird wenn kein fehler zurück kommt. Ich werd ein anders Issue dazu aufmachen und für diese PR um ihn schnell zu mergen das ganze reverten.

Reply:
POST Action calls gehen noch

Reply:
> Das ist best practice Go error handling und findest in jeder codebasis genau so.

Ist mir bekannt. Nicht umsonst gibts fast 150 Proposals zum error handling: https://github.com/golang/go/issues?q=is%3Aissue+is%3Aopen+label%3Aerror-handling

Reply:
nichts ist perfekt. das stimmt. und ich glaube wir können uns in einem issue dazu in ruhe austauschen. Werde mal die Tage eins aufmachen.